panther_hardware_interfaces/include/panther_hardware_interfaces/gpio_controller.hpp

```diff
@@ -0,0 +1,294 @@
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::WATCHDOG, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::AUX_PW_EN, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::CHRG_DISABLE, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::DRIVER_EN, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::E_STOP_RESET, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::FAN_SW, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPOUT1, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPOUT2, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPIN1, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPIN2, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::SHDN_INIT, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::VDIG_OFF, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::VMOT_ON, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{
+      panther_gpiod::GPIOPin::CHRG_SENSE, gpiod::line::direction::INPUT, true},
+  };
+  std::unique_ptr<Watchdog> watchdog_;
+};
+
+class GPIOControllerPTH10X : public GPIOControllerInterface
+{
+public:
+  /**
+   * @brief Initializes the GPIODriver and powers on the motors.
+   */
+  void Start() override;
+
+  /**
+   * @brief Placeholder method indicating lack of hardware E-Stop support for the robot in this
+   * version.
+   *
+   * @return Always returns true.
+   */
+  bool EStopTrigger() override;
+
+  /**
+   * @brief Checks if the motors are powered up (when STAGE2 is active) without controlling any
+   * GPIO.
+   *
+   * @exception std::runtime_error when the motors are not powered up.
+   * @return Always returns true when the motors are powered up.
+   */
+  bool EStopReset() override;
+
+  /**
+   * @brief Controls the motor power by enabling or disabling them based on the 'enable' parameter.
+   *
+   * This method checks if the motors are powered up by verifying the 'STAGE2_INPUT' pin.
+   *
+   * @param enable Set to 'true' to enable the motors, 'false' to disable.
+   * @exception std::runtime_error When attempting to enable the motors without the 'STAGE2_INPUT'
+   * pin active.
+   * @return 'true' if the motor control pin value is successfully set, 'false' otherwise.
+   */
+  bool MotorsEnable(const bool enable) override;
+
+  /**
+   * @brief Placeholder method indicating lack of support for controlling the fan in this robot
+   * version.
+   *
+   * @param enable Ignored parameter in this version.
+   * @exception std::runtime_error Always throws a runtime error due to lack of support for fan
+   * control.
+   */
+  bool FanEnable(const bool /* enable */) override;
```

Comment:
```suggestion
   * @exception std::runtime_error Always throws a runtime error due to lack of support for fan
   * control.
   */
  bool FanEnable(const bool /* enable */) override;
```
Is there a point in putting param here? method definition indicates it is not used. Applies to other methods too

Reply:
It overrides method from base class, so I think that param should stay